server/sandbox_network.go

```diff
@@ -150,20 +135,8 @@ func (s *Server) networkStop(ctx context.Context, sb *sandbox.Sandbox) error {
 	ips := sb.IPs()
 	if len(ips) > 0 {
 		// we only allocated portmappings for the first ip
-		mapping := &hostport.PodPortMapping{
-			Name:         sb.Name(),
-			PortMappings: sb.PortMappings(),
-			HostNetwork:  false,
-		}
-		var err error
-		if utilnet.IsIPv6String(ips[0]) {
-			err = s.hostportManagerv6.Remove(sb.ID(), mapping)
-		} else {
-			err = s.hostportManager.Remove(sb.ID(), mapping)
-		}
-		if err != nil {
-			log.Warnf(ctx, "failed to remove hostport for pod sandbox %s(%s): %v",
-				sb.Name(), sb.ID(), err)
+		if err := s.hostportManager.Remove(sb, ips[0]); err != nil {
```

Comment:
thinking in loud, if we pass the sandbox to the Remove() method we may implement the logic to check if there are still ips inside.

Reply:
I agree with this, I think encapsulating the behavior makes sense :+1: